File: 14-longest-common-prefix/14-longest-common-prefix.scala
import scala.collection.mutable._

object Solution {
    def longestCommonPrefix(strs: Array[String]): String = {
        var done = false
        var res = ""
        var i = 0
        
        while(!done) {
            var set = Set[Char]()
            
            strs.foreach { str =>
                if (str.size > i) set += str.charAt(i)
                else done = true
            }
            
            if (set.size > 1) {
                done = true
            } 
            
            if (!done) {
                res += set.head
                i += 1
            }
        }
        
        res
    }
}

File: 14-longest-common-prefix/14-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        StringBuilder sb = new StringBuilder();
        
        if (strs.length == 0) return sb.toString();
        
        for(int i = 0; i < strs[0].length(); i++) {
            char c = strs[0].charAt(i);
            boolean isSameChar = true;
            
            for(String s : strs) {
                if (s.length() <= i || s.charAt(i) != c) {
                    isSameChar = false;
                    break;
                }
            }
            
            if (isSameChar) sb.append(c);
            else break;
        }
        
        return sb.toString();
    }
}